modules/overrides/config.go

```diff
@@ -0,0 +1,234 @@
+package overrides
+
+import (
+	"flag"
+	"time"
+
+	"github.com/grafana/tempo/pkg/sharedconfig"
+	filterconfig "github.com/grafana/tempo/pkg/spanfilter/config"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/prometheus/common/model"
+)
+
+type ConfigType string
+
+const (
+	ConfigTypeLegacy ConfigType = "legacy"
+	ConfigTypeNew    ConfigType = "new"
+)
+
+const (
+	// LocalIngestionRateStrategy indicates that this limit can be evaluated in local terms only
+	LocalIngestionRateStrategy = "local"
+	// GlobalIngestionRateStrategy indicates that an attempt should be made to consider this limit across the entire Tempo cluster
+	GlobalIngestionRateStrategy = "global"
+
+	// ErrorPrefixLiveTracesExceeded is used to flag batches from the ingester that were rejected b/c they had too many traces
+	ErrorPrefixLiveTracesExceeded = "LIVE_TRACES_EXCEEDED:"
+	// ErrorPrefixTraceTooLarge is used to flag batches from the ingester that were rejected b/c they exceeded the single trace limit
+	ErrorPrefixTraceTooLarge = "TRACE_TOO_LARGE:"
+	// ErrorPrefixRateLimited is used to flag batches that have exceeded the spans/second of the tenant
+	ErrorPrefixRateLimited = "RATE_LIMITED:"
+
+	// metrics
+	MetricMaxLocalTracesPerUser           = "max_local_traces_per_user"
+	MetricMaxGlobalTracesPerUser          = "max_global_traces_per_user"
+	MetricMaxBytesPerTrace                = "max_bytes_per_trace"
+	MetricMaxBytesPerTagValuesQuery       = "max_bytes_per_tag_values_query"
+	MetricMaxBlocksPerTagValuesQuery      = "max_blocks_per_tag_values_query"
+	MetricIngestionRateLimitBytes         = "ingestion_rate_limit_bytes"
+	MetricIngestionBurstSizeBytes         = "ingestion_burst_size_bytes"
+	MetricBlockRetention                  = "block_retention"
+	MetricMetricsGeneratorMaxActiveSeries = "metrics_generator_max_active_series"
+	MetricsGeneratorDryRunEnabled         = "metrics_generator_dry_run_enabled"
+)
+
+var metricLimitsDesc = prometheus.NewDesc(
+	"tempo_limits_defaults",
+	"Default resource limits",
+	[]string{"limit_name"},
+	nil,
+)
+
+type IngestionConfig struct {
+	// Distributor enforced overrides.
+	RateStrategy   string `yaml:"rate_strategy,omitempty" json:"rate_strategy,omitempty"`
+	RateLimitBytes int    `yaml:"rate_limit_bytes,omitempty" json:"rate_limit_bytes,omitempty"`
+	BurstSizeBytes int    `yaml:"burst_size_bytes,omitempty" json:"burst_size_bytes,omitempty"`
+
+	// Ingester enforced overrides.
```

Comment:
I'd keep the original comments: these are still actual limits we enforce, these limits are just part of the overrides (= properties we can configure at run-time).

```suggestion
	// Distributor enforced limits.
	RateStrategy   string `yaml:"rate_strategy,omitempty" json:"rate_strategy,omitempty"`
	RateLimitBytes int    `yaml:"rate_limit_bytes,omitempty" json:"rate_limit_bytes,omitempty"`
	BurstSizeBytes int    `yaml:"burst_size_bytes,omitempty" json:"burst_size_bytes,omitempty"`

	// Ingester enforced limits.
```

Reply:
Right. Done in 0de405260025ee7b3eb7ab66339f73a05d0a459c